### PR TITLE
feat(vault): Added vault configuration endpoint PUT

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Added vault configuration endpoint PUT

**Fixes** # MON-14984

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Security/Vault/Application/UseCase/UpdateVaultConfiguration/UpdateVaultConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    public function __construct(
+        private ReadVaultConfigurationRepositoryInterface $readVaultConfigurationRepository,
+        private WriteVaultConfigurationRepositoryInterface $writeVaultConfigurationRepository,
+        private ReadVaultRepositoryInterface $readVaultRepository,
+        private VaultConfigurationFactory $factory,
+        private ContactInterface $user
+    ) {
+    }
+
+    public function __invoke(
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Application/UseCase/UpdateVaultConfiguration/UpdateVaultConfiguration.php:64]: phpdoc

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

File: centreon/src/Core/Security/Vault/Infrastructure/API/CreateVaultConfiguration/CreateVaultConfigurationController.php [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 declare(strict_types=1);
 
-namespace Core\Security\Vault\Infrastructure\Api\CreateVaultConfiguration;
+namespace Core\Security\Vault\Infrastructure\API\CreateVaultConfiguration;
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Infrastructure/API/CreateVaultConfiguration/CreateVaultConfigurationController.php:24]: 🚨 A t the moment, we have **`API`** and **`Api`** namespaces (the second seems to be more used)
👉️ We need to fix a "norm" for that.

File: centreon/src/Core/Security/Vault/Infrastructure/Repository/DbReadVaultConfigurationRepository.php [hunk 3]
```diff
@@ -54,26 +54,73 @@ public function findByAddressAndPortAndStorage(
     ): ?VaultConfiguration {
         $this->info('Getting existing vault configuration by address, port and storage');
 
-        $record = [];
         $statement = $this->db->prepare(
             $this->translateDbName(
-                'SELECT * FROM `:db`.`vault_configuration` WHERE `url`=:address AND `port`=:port AND `storage`=:storage'
+                <<<SQL
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Infrastructure/Repository/DbReadVaultConfigurationRepository.php:59]: php-cs-fixer needs this I think.
```suggestion
                <<<'SQL'
```

File: centreon/src/Core/Security/Vault/Infrastructure/Repository/DbWriteVaultConfigurationRepository.php [hunk 4]
```diff
@@ -45,22 +46,59 @@ public function __construct(DatabaseConnection $db)
...
+    /**
+     * @inheritDoc
+     */
+    public function update(VaultConfiguration $vaultConfiguration): void
+    {
+        $this->info('Updating vault configuration in database');
+
+        $statement = $this->db->prepare(
+            $this->translateDbName(
+                <<<SQL
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Infrastructure/Repository/DbWriteVaultConfigurationRepository.php:84]: Same as previous
```suggestion
                <<<'SQL'
```

File: centreon/src/Core/Security/Vault/Infrastructure/Repository/DbWriteVaultConfigurationRepository.php [hunk 5]
```diff
@@ -45,22 +46,59 @@ public function __construct(DatabaseConnection $db)
...
     {
-        $this->info('Adding new vault configuration');
+        $this->info('Adding new vault configuration in database');
 
         $statement = $this->db->prepare(
             $this->translateDbName(
-                'INSERT INTO `:db`.`vault_configuration` (`name`, `type_id`, `url`, `port`, `storage`, `role_id`, '
-                    . '`secret_id`, `salt`) VALUES (:name, :type_id, :url, :port, :storage, :role_id, :secret_id, '
-                    . ':salt)'
+                <<<SQL
```
Comment (abuathier) [inline centreon/src/Core/Security/Vault/Infrastructure/Repository/DbWriteVaultConfigurationRepository.php:56]: Same as previous
```suggestion
                <<<'SQL'
```

**Review by abuathier — APPROVED** [review]

File: centreon/src/Core/Security/Vault/Infrastructure/API/CreateVaultConfiguration/CreateVaultConfigurationController.php (hunk 2 above)
Comment (dmyios) [reply centreon/src/Core/Security/Vault/Infrastructure/API/CreateVaultConfiguration/CreateVaultConfigurationController.php:24]: The change of namespace was proposed by Laurent as it is rather an abbreviation than a name. To be discussed

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**